File: apple/Sources/ReactNativeWatchOS/Modifiers/Offset.swift
import SwiftUI

/// SwiftUI `.offset(x:y:)`. Shifts the view after layout without affecting
/// the positions of surrounding views.
enum RNWOffsetModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("offset") { view, params, _ in
            AnyView(view.offset(
                x: params.cgFloat("x") ?? 0,
                y: params.cgFloat("y") ?? 0
            ))
        }
    }
}

File: src/modifiers/offset.ts
import { createModifier } from './createModifier';

export interface OffsetParams {
  /// Horizontal offset in points. Positive moves the view trailing.
  x?: number;
  /// Vertical offset in points. Positive moves the view down.
  y?: number;
}

/// SwiftUI `.offset(x:y:)`. Shifts the view by the given amount *after*
/// layout, so surrounding views keep their original positions.
export function offset(params: OffsetParams): ReturnType<typeof createModifier>;
export function offset(
  x: number,
  y: number
): ReturnType<typeof createModifier>;
export function offset(a: OffsetParams | number, b?: number) {
  if (typeof a === 'number') {
    return createModifier('offset', { x: a, y: b ?? 0 });
  }
  return createModifier('offset', a);
}
